src/contracts/factory/FundFactory.ts

```diff
@@ -31,6 +80,167 @@ export class FundFactory extends Contract {
+      const registry = new Registry(this.environment, registryAddress);
+      if (!(await registry.isAssetRegistered(args.denominationAsset))) {
+        throw new DenominationAssetNotRegisteredError();
+      }
+    };
+
+    return this.createTransaction({ from, method, methodArgs, validate });
+  }
+
+  public createAccouting(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).accounting);
+    };
+
+    return this.createTransaction({ from, method: 'createAccouting', validate });
+  }
+
+  public createFeeManager(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).accounting);
+    };
+
+    return this.createTransaction({ from, method: 'createFeeManager', validate });
+  }
+
+  public createParticipation(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).participation);
+    };
+
+    return this.createTransaction({ from, method: 'createParticipation', validate });
+  }
+
+  public createPolicyManager(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).policyManager);
+    };
+
+    return this.createTransaction({ from, method: 'createPolicyManager', validate });
+  }
+
+  public createShares(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).shares);
+    };
+
+    return this.createTransaction({ from, method: 'createShares', validate });
+  }
+
+  public createTrading(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).trading);
+    };
+
+    return this.createTransaction({ from, method: 'createTrading', validate });
+  }
+
+  public createVault(from: Address) {
+    const validate = async () => {
+      this.validateComponentSet(await this.getManagersToHubs(from));
+      this.validateComponentNotSet((await this.getManagersToRoutes(from)).vault);
+    };
+
+    return this.createTransaction({ from, method: 'createVault', validate });
+  }
+
+  public completeSetup(from: Address) {
+    const validate = async () => {
+      const hub = await this.getManagersToHubs(from);
+      if (await this.isInstance(hub)) {
+        throw new SetupAlreadyComplete();
+      }
+
+      const routes = await this.getManagersToRoutes(from);
+
+      if (
+        !this.componentExists(hub) ||
```

Comment:
Instead of checking all in sequence and then not returning a specific error, I'd simply re-use the `validateRoutesSet` method and throw an explicit error message at the first failing condition.